src/material/form-field/testing/form-field-harness.ts

```diff
@@ -137,7 +137,11 @@ export abstract class _MatFormFieldHarnessBase<
   /** Gets hint messages which are currently displayed in the form-field. */
   async getTextHints(): Promise<string[]> {
     const hints = await this._hints();
-    return parallel(() => hints.map(e => e.text()));
+    const hintsOrNullIfHidden = await parallel(() =>
```

Comment:
Rather than checking the computed `display`, we can change the selector to something like `.mat-form-field-hint-wrapper:not(.mat-mdc-form-field-hint-wrapper-inactive)`. Same goes for the other places where we're querying for visible hints.

Reply:
Wouldn't it be preferable to use the computed `display` since if the class name is changed the tests will incorrectly pass? And the intent of the tests is to ensure the hints are not displayed.

Reply:
Just bumping on this. :)

Reply:
We're the ones in control of the class name so we can update the tests if the class name changes.